packages/web/app/features/ProgramRegistry/PatientProgramRegistryUpdateFormModal.jsx

```diff
@@ -0,0 +1,337 @@
+    for (const [group, conditions] of Object.entries(groupMapping)) {
+      if (conditions.includes(conditionCategory)) {
+        groupedData[group].push({
+          conditionId: programRegistryCondition.id,
+          name: programRegistryCondition.name,
+          date,
+          conditionCategory,
+        });
+        break;
+      }
+    }
+  });
+  Object.keys(groupedData).forEach(group => {
+    groupedData[group].sort((a, b) => a.name?.localeCompare(b?.name));
+  });
+  return groupedData;
+};
+
+export const PatientProgramRegistryUpdateFormModal = ({
+  patientProgramRegistration = {},
+  onClose,
+  open,
+}) => {
+  const { programRegistryId, patientId, clinicalStatusId } = patientProgramRegistration;
+  const { mutateAsync: submit, isLoading: isSubmitting } = useUpdateProgramRegistryMutation(
+    patientId,
+  );
+  const { data: conditions = [], isLoading } = usePatientProgramRegistryConditionsQuery(
+    patientId,
+    programRegistryId,
+  );
+  const programRegistryStatusSuggester = useSuggester('programRegistryClinicalStatus', {
+    baseQueryParameters: { programRegistryId },
+  });
+
+  const handleSubmit = async data => {
+    await submit(data);
+    onClose();
+  };
+
+  if (!patientProgramRegistration) return null;
+
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <Modal
+      title={
+        <TranslatedText
+          stringId="patientProgramRegistry.updateModal.title"
+          fallback="Update program registry"
+        />
+      }
+      open={open}
+      onClose={onClose}
+      width="lg"
+    >
+      <Form
+        showInlineErrorsOnly
+        onSubmit={handleSubmit}
+        render={({ setFieldValue, values, initialValues, dirty }) => {
+          const groupedData = values.conditions;
+          const getIsDirty = index => {
+            return (
+              values.conditions[index]?.conditionCategory !==
+              initialValues.conditions[index]?.conditionCategory
+            );
+          };
+
+          console.log('values', values);
+
+          const columns = [
+            {
+              key: 'condition',
+              title: (
+                <span id="condition-label">
+                  <TranslatedText
+                    stringId="patientProgramRegistry.updateConditionModal.condition"
+                    fallback="Condition"
+                  />
+                </span>
+              ),
+              accessor: ({ name }, groupName, index) => {
+                if (name) {
+                  return name;
+                }
+
+                const onClear = () => {
+                  console.log('clearing');
+                };
+
+                const isLastRow = index === groupedData.confirmedSection.length - 1;
+
+                return (
+                  <div style={{ position: 'relative' }}>
+                    <ProgramRegistryConditionField
+                      name={`conditions[${groupName}][${index}].conditionId`}
```

Comment:
I structure the form fields like this so that the form values align with the grouped data structure and that way I don't have to do transformations between the form and the data.